pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -228,6 +238,9 @@ public BrokerResponse handleRequest(JSONObject request, @Nullable RequesterIdent
     Map<String, List<String>> offlineRoutingTable = null;
     Map<String, List<String>> realtimeRoutingTable = null;
     if (offlineBrokerRequest != null) {
+      if (realtimeBrokerRequest == null) {
+        requestStatistics.setFanoutType(RequestStatistics.FanoutType.OFFLINE);
```

Comment:
It looks like this can be set in the if-else block of line 222?

Reply:
yes. Changed.